src/main/java/ui/UI.java

```diff
@@ -150,6 +155,7 @@ private void disableUI(boolean disable) {
     }
 
     private void showMainWindow(String repoId) {
+        boolean isAFirstTimeUser = logic.getStoredRepos().isEmpty();
```

Comment:
I replied that this line cannot be moved down. `logic.openPrimaryRepository` changes `logic.getStoredRepos()` and hence the check has to be made exactly at this point. The boolean value obtained here is then used to decide whether the sample board should be opened.